src/main/java/com/aws/greengrass/componentmanager/ComponentManager.java

```diff
@@ -206,6 +206,15 @@ private ComponentIdentifier negotiateVersionWithCloud(String componentName,
                 resolvedComponentVersion = RetryUtils.runWithRetry(clientExceptionRetryConfig,
                         () -> componentServiceHelper.resolveComponentVersion(componentName, null, versionRequirements),
                         "resolve-component-version", logger);
+
+                String vendorGuidance = resolvedComponentVersion.vendorGuidanceAsString();
+                if (vendorGuidance != null && vendorGuidance.equals("DISCONTINUED")) {
```

Comment:
don't we also have a string that tells the customer what they should use instead? We should log that out.

Reply:
not sure about a string which tells what to use instead, followed the description of DISCONTINUED [here](https://docs.aws.amazon.com/greengrass/v2/APIReference/API_ResolvedComponentVersion.html)

Reply:
using vendor guidance enum now to compare discontinued component